### ISSUE TITLE
[Question] How to get a slice of bytes of websocket blob?

### ISSUE THREAD

**Issue by Unmoving** [koute/stdweb#140]
Hello!

I try to send bytes encoded by bincode through the websocket. On client side I would like to deserialize the data, but for that I need something like:

```
ws.add_event_listener(|event: SocketMessageEvent| {
   let decoded_message: Message = deserialize(&event.data()[..]).unwrap();
   alert(&format!{"Message from WS: {:?}", decoded_message});
});
```

This gives 

> cannot index into a value of type `stdweb::webapi::events::socket::SocketMessageData`

So I need some way to slice into the actual data. Any nice approaches I miss?

I tried with:

```
let decoded_message: Message = deserialize(&event.data().into_array_buffer().unwrap()[..]).unwrap();
```

and same for blob.

After having a look at the code, I can't even get access to the underlying Reference.

Maybe my usecase is special, but could you point me in the right direction, please?

I know (or think that I do at least) how to do it in plain js, but that's something I would like to avoid.

**Comment by koute** [comment]
Ideally you'd use `set_binary_type` on the `WebSocket` instance to have it give you a typed array, and then you could convert that array to `Vec<u8>`. Unfortunately right now the `BinaryType` type is not exported (I'll fix it in `0.4.1`) so you can't use it; as a workaround you can set the binary type manually through the `js!` macro.

**Comment by Unmoving** [comment]
Just for the record (in case anyone ever stumbles upon this), it's now as easy as:

`ws.set_binary_type(SocketBinaryType::ArrayBuffer);`
and then:
```
 let data: Vec<u8> = Vec::from(event.data().into_array_buffer().unwrap());
 let decoded_message: Message = deserialize(&data[..]).unwrap();
```
